cmake/kokkos_arch.cmake

```diff
@@ -144,6 +144,11 @@ ENDIF()
 GLOBAL_SET(KOKKOS_AMDGPU_OPTIONS)
 IF(KOKKOS_CXX_COMPILER_ID STREQUAL HIP)
   SET(AMDGPU_ARCH_FLAG "--amdgpu-target")
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL Cray)
+  SET(AMDGPU_ARCH_FLAG "--cuda-gpu-arch")
```

Comment:
I believe `--offload-arch` should be a generic synonym for `--cuda-gpu-arch` in all Clang variants now.

Reply:
Yes.  If I recall correctly we still need `--amdgpu-target` in the HIP branch pre ROCm 4.0
We also need to strip the flag when we call the linker because we get a warning.